src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -20,8 +21,48 @@ public interface INodeHandler : IHandler
         BaseNodeHandlerData Data { get; set; }
     }
 
+    [ServiceLocator(Default = typeof(NodeHandlerHelper))]
+    public interface INodeHandlerHelper
+    {
+        string[] GetFilteredPossibleNodeFolders(string nodeFolderName, string[] possibleNodeFolders);
+        string GetNodeFolderPath(string nodeFolderName, IHostContext hostContext);
+        bool IsNodeFolderExist(string nodeFolderName, IHostContext hostContext);
+    }
+
+    public class NodeHandlerHelper : INodeHandlerHelper
+    {
+        public bool IsNodeFolderExist(string nodeFolderName, IHostContext hostContext) => File.Exists(GetNodeFolderPath(nodeFolderName, hostContext));
+
+        public string GetNodeFolderPath(string nodeFolderName, IHostContext hostContext) => Path.Combine(
+            hostContext.GetDirectory(WellKnownDirectory.Externals),
+            nodeFolderName,
+            "bin",
+            $"node{IOUtil.ExeExtension}");
+
+        public string[] GetFilteredPossibleNodeFolders(string nodeFolderName, string[] possibleNodeFolders)
+        {
+            for (int i = 0; i < possibleNodeFolders.Length; i++)
+            {
+                if (possibleNodeFolders[i] == nodeFolderName)
+                {
+                    return possibleNodeFolders.Skip(i + 1).ToArray();
+                }
+            }
+
+            return Array.Empty<string>();
+        }
```

Comment:
If I understand the logic correctly, it can be written in more readable way
```
var nodeFolderIndex = possibleNodeFolders.ToList().IndexOf(nodeFolderName);

return nodeFolderIndex >= 0 ? 
    possibleNodeFolders.Skip(nodeFolderIndex + 1).ToArray() 
    : Array.Empty<string>();
```

Reply:
Thank you, I have refactored the part of code